### PR TITLE
Media upload error codes self-hosted in media lib

### PR DESCRIPTION
**Pull request by develric**
This PR is part of a series related to #12325

The changes here should address the Media Library with self-hosted scenario. The equivalent scenarios for GB Editor will be covered in subsequent PRs.

This has a companion FluxC PR [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1658)

The Fluxc modifications map the 401 and 500 upload media errors that can be found [here](https://github.com/WordPress/WordPress/blob/649ab58d96491549f8b617d2702bb437488bb16d/wp-includes/class-wp-xmlrpc-server.php#L6298)

NOTE: I was trying to use a similar approach as we did in #12620 ; that is leveraging the user error message that the API is actually able to provide and display them to the user. Unfortunately AFAIU in the case of XML-RPC API we do not send or use a `locale`parameter that the client app can send to the server to declare it's idiom. The only thing I was able to get is a localized message changing the language server side, but that is not a per user triggered localization so we cannot use this approach. I reverted to using general string resources that only map the category of the error (401 and 500).

### To test
Access to a pure self-hosted site in the app.

- Open the Media Library screen and try to upload a file
- Force trigger the errors as per the link above (I can provide access to a test site I have setup for that or you can use Charles proxy rewrite feature or AS debugger for this)
- Check you get the relevant string resources message in the snackbar

NOTE: if you are using the Charles proxy approach, the message received from the server should have a 200 http status and the following xml content:

#### 401 case
```
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>401</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>Sorry, you are not allowed to upload files.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
```

#### 500 case
```
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>500</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>Could not write file.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
```

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1594,6 +1594,8 @@
     <string name="error_media_canceled">Uploading media were canceled</string>
     <string name="error_media_quota_exceeded">Storage quota exceeded</string>
     <string name="error_media_quota_exceeded_toast">Cannot upload file.\nStorage quota was exceeded.</string>
+    <string name="error_media_xmlrpc_not_allowed">Operation not allowed</string>
```
Comment (planarvoid) [inline WordPress/src/main/res/values/strings.xml:1600]: Could we maybe make this a bit more explicit? Maybe something like "Operation not allowed by XMLRPC"?

**Review by planarvoid — COMMENTED** [review]
I've tested both the cases with Charles rewrite and they show the correct error 👍 . I have one suggestion on the FluxC PR. Let me know what you think.

**Comment by develric** [comment]
Moving the milestone to 15.7 since makes sense to take care of [this](https://github.com/wordpress-mobile/WordPress-Android/pull/12672#issuecomment-674990529) design improvements before to ship these chained PRs.

**Comment by develric** [comment]
Hi @planarvoid , just noting that I'm addressing the copy change to `error_media_xmlrpc_not_allowed` string in #12672 since all discussion about copy change happened in that PR.

**Review by planarvoid — APPROVED** [review]
Looks good 👍  thanks!

**Pull request merged**